From cbe6d682a5cb7d09a6ced33dc84fd43ce9d21c7b Mon Sep 17 00:00:00 2001
From: Pierre Ossman <ossman@cendio.se>
Date: Tue, 1 Apr 2025 16:47:44 +0300
Subject: [PATCH 2/2] libltdl: Fix libltdl early failures for multi-arch

Since there could be multiple files for libltdl to try dlopening,
libltdl should not fail after the first attempted file.

* libltdl/ltdl.c: Change return if dlopen() fails.
* libltdl/ltdl.mk: Update serial.
---
 libltdl/ltdl.c  | 5 ++---
 libltdl/ltdl.mk | 2 +-
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/libltdl/ltdl.c b/libltdl/ltdl.c
index 74bfe431..60495415 100644
--- a/libltdl/ltdl.c
+++ b/libltdl/ltdl.c
@@ -785,10 +785,9 @@ find_handle_callback (char *filename, void *data, void *data2)
   if (notfound)
     return 0;
 
-  /* Try to dlopen the file, but do not continue searching in any
-     case.  */
+  /* Try to dlopen the file. */
   if (tryall_dlopen (phandle, filename, advise, 0) != 0)
-    *phandle = 0;
+    return 0;
 
   return 1;
 }
diff --git a/libltdl/ltdl.mk b/libltdl/ltdl.mk
index 46109775..0cb427d8 100644
--- a/libltdl/ltdl.mk
+++ b/libltdl/ltdl.mk
@@ -35,7 +35,7 @@ AM_CPPFLAGS	       += -DLT_CONFIG_H='<$(LT_CONFIG_H)>' \
 			  -I$(srcdir)/libltdl -Ilibltdl/libltdl \
 			  -I$(srcdir)/libltdl/libltdl
 AM_LDFLAGS	       += -no-undefined
-LTDL_VERSION_INFO	= -version-info 10:4:3
+LTDL_VERSION_INFO	= -version-info 10:5:3
 
 noinst_LTLIBRARIES	+= $(LT_DLLOADERS)
 
-- 
2.45.2

